test/runtests.jl

```diff
@@ -11,3 +11,8 @@ using Unitful.DefaultSymbols
 
 include("F64.jl")
 include("test_dists.jl")
+
+# Test ChainRules definitions on Julia versions that support weak dependencies
+if isdefined(Base, :get_extension)
+    include("chainrules.jl")
+end
```

Comment:
`Base.get_extension` is documented as internal so it could be dropped and we wouldn't notice that the test isn't run.
```suggestion
if VERSION >= v"1.10.0-DEV"
    include("chainrules.jl")
end
```

Reply:
That seems wrong, it definitely exists in Julia 1.9.

While I can see the general point here, I think it's a rather hypothetical issue since the official way to detect whether weak dependencies are supported is this check (see, e.g., https://pkgdocs.julialang.org/v1/creating-packages/#Backwards-compatibility).

Reply:
BTW where is it documented as internal? In the docs (https://docs.julialang.org/en/v1/base/base/#Base.get_extension) it is not listed in the section "Internals" but "Code loading" it seems.

Reply:
Oh wow, I checked the version of the relevant commit (https://github.com/JuliaLang/julia/commit/495a004bda33e284b0acc612f5ced9ba1eb9a777), and it is `1.10.0-DEV.109` even though it exists in 1.9. Apparently, the feature was only backported to 1.9.

Reply:
But adding something like
```julia
if VERSION >= v"1.10.0-DEV.109"
    include("chainrules.jl")
end
```
would mean that these tests don't run on 1.9, wouldn't it? But we definitely want to test 1.9.

Reply:
😬 do we know what the way forward with this is?

Reply:
Maybe we could change it to
```suggestion
# The VERSION check ensures that the tests are run even if
# Base.get_extension is removed
# Ref: https://github.com/JuliaStats/Distances.jl/pull/246#discussion_r1334654115
if VERSION >= v"1.10.0-DEV.109" || isdefined(Base, :get_extension)
    include("chainrules.jl")
end
```
@nalimilan, what do you think?

Reply:
I don't understand the issue if `isdefined(Base, :get_extension)` is the documented procedure to do this?

Reply:
Just making suggestions for how to resolve the discussions, I don't have a strong opinion regarding whether an additional check should be added or not 🙂 In any case, I think it just depends on what @nalimilan thinks.

Reply:
I appreciate the caution but this PR seems to be stuck for a long time for no good reason. @nalimilan already approved the PR, so the comment above is likely not a blocker for him.

Reply:
Given for how long the PR has been open and that @nalimilan was the first and main reviewer of the PR, I think there's no harm in giving him time to reply. I plan to merge end of next week if he hasn't disapproved explicitly by then.

Reply:
Sorry. Yeah, I approved so that you can be free to merge as you like.

I think the commit on 1.9 is 327e0815d6dc7efef0cfaf82c2d2e9c01d55e304. `if VERSION >= v"1.9.0-alpha1.50"` should work on both 1.9 and 1.10.

Reply:
Ah, thanks for the pointer! Basing it on the backport commit is the cleanest way, of course, if we want to change it to a `VERSION` check. It seems the relevant backport commit that introduced `Base.get_extension` to 1.9 was actually https://github.com/JuliaLang/julia/commit/93587d7c1015efcd4c5184e9c42684382f1f9ab2 (the other one is "just" an improvement/bugfix for package extensions). In practice it won't matter but so I'll go with
```suggestion
# Support for extensions was added in
# https://github.com/JuliaLang/julia/commit/93587d7c1015efcd4c5184e9c42684382f1f9ab2
# https://github.com/JuliaLang/julia/pull/47695 
if VERSION >= v"1.9.0-alpha1.18"
    include("chainrules.jl")
end
```